### ISSUE TITLE
Mispellings across code

### ISSUE THREAD

**Issue by Benjamin-McRae-Tracsis** [haskell/aeson#1116]
`unespected` is a spelling error in at least 4 places I can see, plus tests.

Here is an example: https://github.com/haskell/aeson/blob/c93d60d82388c184a9cf240f016a728c9d0e758f/src/Data/Aeson/Decoding/ByteString.hs#L164C19-L164C29

It would be good to remove such spelling errors.

**Comment by ad-si** [comment]
You could use https://github.com/crate-ci/typos. It works surprisingly well!

**Comment by dhruvdabhi101** [comment]
Hey, i would like to work on this if it is open to work.

**Comment by L0neGamer** [comment]
Looks like someone was fixing it in #1081 before I even made this issue. I assume that @phadej would prefer to merge that than make a new PR.

**Comment by phadej** [comment]
Yes, but we are careful not to change error messages in minor releases, as something (usually tests) may break somewhere. But alone spelling errors don't justify a major release, so I'm stalling on this issue.
